modules/ting_search_context/ting_search_context.admin.inc

```diff
@@ -0,0 +1,580 @@
+ */
+
+/**
+ * Page callback (admin/content/ting-search-context).
+ *
+ * Renders the rate nodes VBO view.
+ */
+function ting_search_context_admin_rate_nodes() {
+  $output = t("The rate nodes view wasn't found");
+  if ($view = views_get_view('ting_search_context_rate_nodes')) {
+    $output = $view->preview('block_1');
+  }
+  return $output;
+}
+
+/**
+ * Callback for the context overview administratition page.
+ */
+function ting_search_context_admin_contexts() {
+  $output = array();
+  if ($view = views_get_view('ting_search_context_contexts')) {
+    $output = $view->preview('block_1');
+  }
+  return $output;
+}
+
+/**
+ * Form-builder for the context form. Used when adding and editing contexts.
+ *
+ * @see ting_search_context_forms()
+ * @see ting_search_context_admin_context_form_validate()
+ * @see ting_search_context_admin_context_form_submit()
+ * @see ting_search_context_admin_context_form_cancel()
+ */
+function ting_search_context_admin_context_form($form, &$form_state, $context = FALSE) {
+  if ($context) {
+    // Save the context object in form_state for later use.
+    $form_state['ting_search_context'] = $context;
+  }
+
+  $form['add_context'] = array(
+    '#type' => 'fieldset',
+    '#title' => t('Add context'),
+  );
+  $types = ting_search_context_get_types();
+  // We don't allow creation or editing of system context types.
+  if (!$context) {
+    unset($types['system']);
+  }
+  $form['add_context']['type'] = array(
+    '#type' => 'select',
+    '#title' => t('Context type'),
+    '#description' => t('Select the type of context to create'),
+    '#options' => $types,
+    '#default_value' => $context ? $context->type : 'search',
+    '#disabled' => $context && $context->type == 'system',
+  );
+  $form['add_context']['name'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Name'),
+    '#description' => t('Enter the name of the context'),
+    '#maxlength' => 255,
+    '#required' => TRUE,
+    '#default_value' => $context ? $context->name : '',
+    '#disabled' => $context && $context->type == 'system',
+  );
+  $form['add_context']['search'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Search string'),
+    '#description' => t('Enter the search string'),
+    '#maxlength' => 255,
+    '#states' => array(
+      'visible' => array(
+        ':input[name="type"]' => array('value' => 'search'),
+      ),
+      'required' => array(
+        ':input[name="type"]' => array('value' => 'search'),
+      ),
+    ),
+    '#default_value' => $context && $context->type == 'search' ? $context->search : '',
+  );
+  $form['add_context']['context'] = array(
+    '#type' => 'select',
+    '#title' => t('Parent context'),
+    '#description' => t('Select the parent context'),
+    '#options' => array(
+      'voksen_fag' => t('Voksen faglitteratur'),
+      'voksen_skøn' => t('Voksen skønlitteratur'),
+      'børne_fag' => t('Børne faglitteratur'),
+      'børne_skøn' => t('Børne skønlitteratur'),
```

Comment:
All default text content in modules must be in English (See: Code guidelines,  [Code Structure] (https://github.com/ding2/ding2/blob/master/docs/code_guidelines.md#code-structure)). Also, non-english option values is bad idea.